src/extensibility/ExtensionManagerDialog.js

```diff
@@ -107,57 +110,98 @@ define(function (require, exports, module) {
                     .text(Strings.PROCESSING_EXTENSIONS)
                     .append("<span class='spinner inline spin'/>");
                 
-                ExtensionManager.removeMarkedExtensions()
+                var removeExtensionsPromise,
+                    updateExtensionsPromise,
+                    disableExtensionsPromise,
+                    removeErrors,
+                    updateErrors,
+                    disableErrors;
+                
+                removeExtensionsPromise = ExtensionManager.removeMarkedExtensions();
+                removeExtensionsPromise
```

Comment:
`.fails` returns a promise, so you could remove this promise repetitions and chain with the assignment.